src/cryptonote_basic/miner.cpp

```diff
@@ -835,6 +835,12 @@ namespace cryptonote
 
     #elif defined(__APPLE__)
         
+      if(&IOPSGetTimeRemainingEstimate == NULL)
```

Comment:
Yes this should work according to Apples docs because frameworks symbols are marked as weakly linked which means at runtime the function will either be resolved or null, hence the null check. 

Unfortunately I do not have access to a machine running < 10.7 so I cannot properly test this though. 

The other option is to use Apples "availability macros" at compile time but this would mean binaries compiled on < 10.7 would not have this feature if run on 10.7+ machines.